pkg/apis/crds/karpenter.azure.com_aksnodeclasses.yaml

```diff
@@ -312,3 +312,295 @@ spec:
     storage: true
     subresources:
```

Comment:
Should we set the `deprecated` field too?

Reply:
I was wondering whether we do that as a separate PR. defer to @tallaxes

Reply:
No strong opinion, as long as we do it (would want to have it in the release), and make sure that version-sensitive tests pass when we do.

Reply:
This should work:
```go
// +kubebuilder:deprecatedversion:warning="use v1beta1.AKSNodeClass instead"
```

Reply:
Added v1alpha2 deprecation